src/NavBar.js

```diff
@@ -324,9 +325,11 @@ class NavBar extends React.Component {
 
   renderTitle(childState, index:number) {
     const title = this.props.getTitle ? this.props.getTitle(childState) : childState.title;
+    const numberOfLines = this.props.titleNumberOfLines ? this.props.titleNumberOfLines : -1;
```

Comment:
In Javascript `false == 0`, so if `this.props.titleNumberofLines` is defined to `0`, it will be defaulted to `-1` by this line.

In case is what you want, I suggest this defaulting instead :

```javascript
const numberOfLines = this.props.titleNumberOfLines || -1;
```

Otherwise : 
```javascript
const numberOfLines = Number.isInteger(this.props.titleNumberOfLines) ? this.props.titleNumberOfLines : -1;
```